src/main/java/billgates/usecases/user_join/UserJoinUseCase.java

```diff
@@ -0,0 +1,35 @@
+package billgates.usecases.user_join;
+
+import billgates.database.QueryUserData;
+import billgates.interface_adapters.DatabaseGateway;
+
+import java.util.List;
+
+public class UserJoinUseCase implements UserJoinInputPort {
+    private final DatabaseGateway gateway;
+    private final UserJoinOutputPort outputPort;
+
+    public UserJoinUseCase(DatabaseGateway gateway, UserJoinOutputPort outputPort) {
+        this.gateway = gateway;
+        this.outputPort = outputPort;
+    }
+
+    @Override
+    public void join(UserJoinRequestModel model) {
+        QueryUserData userData = gateway.getUserData();
+        if (!userData.getUsers().contains(model.getUsername())) {
+            // TODO add the user in the database.
+
+            // TODO log the user in
```

Comment:
And then we need to pop a dialog to notify the user that he/she has successfully registered.